### PR TITLE
Various fixes to VSCode extension

### PR DESCRIPTION
**Pull request by yannham**
This PR packs a set of small modifications and fixes before publishing the VSCode extension on the VSCode market place:

 - Rename the parent containing directory from client-extension to vscode-extension
 - Add a display name and a description
 - Update the README, add links to non-Nix installation methods
 - Small fixes to the TextMate Nickel grammar

### CODE REVIEW

**Comment by yannham** [comment]
@szlend, if you wouldn't mind taking a quick look, I will be reassured, being neither a VSCode user, let alone a VSCode developer.

File: lsp/vscode-extension/language-configuration.json [hunk 1]
```diff
@@ -1,32 +1,26 @@
...
     ["(", ")"]
   ],
-  // symbols that are auto closed when typing
   "autoClosingPairs": [
     ["{", "}"],
     ["[", "]"],
     ["[|", "|]"],
     ["(", ")"],
-    ["\"", "\""],
-    ["'", "'"]
```
Comment (yannham) [inline lsp/vscode-extension/language-configuration.json:21]: Single quotes are used for enum tags in Nickel, like quoting in Lisp, that is `some_function 'Http 'Json`, but isn't a pair opening/closing delimiter.

File: lsp/vscode-extension/syntaxes/nickel.tmLanguage.json [hunk 2]
```diff
@@ -45,16 +45,16 @@
...
       "patterns": [
         {
           "name": "constant.character.escape.interpolation.nickel",
-          "match": "''%$\\{"
+          "match": "\\\\1$\\{"
         },
         {
           "name": "constant.other.placeholder.nickel",
-          "begin": "%\\{",
+          "begin": "\\1\\{",
```
[the remark this answers is not in the document]
Comment (szlend) [reply lsp/vscode-extension/syntaxes/nickel.tmLanguage.json:57]: Hm, I don't think string interpolation works at all. I just checked out master and it didn't work there either, so it's not like this change broke it. It also doesn't escape `"`. Tbh I don't really understand `tmLanguage` very well, but I can see if I can make it work somehow. Personally I don't mind if the first version isn't absolutely perfect though.

File: lsp/vscode-extension/language-configuration.json (hunk 1 above)
Comment (szlend) [reply lsp/vscode-extension/language-configuration.json:21]: Hm, should we make it some form of keyword/operator then?

File: lsp/vscode-extension/syntaxes/nickel.tmLanguage.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (szlend) [reply lsp/vscode-extension/syntaxes/nickel.tmLanguage.json:57]: I think this whole thing is completely wrong tbh. The thing called `strings_interpolation` should handle `%{}`. Then we need to introduce a `multiline_string` that handles `m%""%`. Inside `string` and `multiline_string` we should include `strings_interpolation`. And `strings_interpolation` should include `expression`. What I call `expression` should be any Nickel expression (keywords, strings, operators, etc.). This should make it recursive.

[the remark this answers is not in the document]
Comment (szlend) [reply lsp/vscode-extension/syntaxes/nickel.tmLanguage.json:57]: This is pretty close for example:
```
diff --git a/lsp/vscode-extension/syntaxes/nickel.tmLanguage.json b/lsp/vscode-extension/syntaxes/nickel.tmLanguage.json
index 1d82526b..ec85d9f9 100644
--- a/lsp/vscode-extension/syntaxes/nickel.tmLanguage.json
+++ b/lsp/vscode-extension/syntaxes/nickel.tmLanguage.json
@@ -3,34 +3,41 @@
   "name": "Nickel",
   "patterns": [
     {
-      "include": "#keywords"
-    },
-    {
-      "include": "#strings"
-    },
-    {
-      "include": "#comments"
-    },
-    {
-      "include": "#operators"
-    },
-    {
-      "include": "#numbers"
-    },
-    {
-      "include": "#bools"
-    },
-    {
-      "include": "#storage"
-    },
-    {
-      "include": "#types"
-    },
-    {
-      "include": "#strings_interpolation"
+      "include": "#expression"
     }
   ],
   "repository": {
+    "expression": {
+      "patterns": [
+        {
+          "include": "#keywords"
+        },
+        {
+          "include": "#multiline_strings"
+        },
+        {
+          "include": "#strings"
+        },
+        {
+          "include": "#comments"
+        },
+        {
+          "include": "#operators"
+        },
+        {
+          "include": "#numbers"
+        },
+        {
+          "include": "#bools"
+        },
+        {
+          "include": "#storage"
+        },
+        {
+          "include": "#types"
+        }
+      ]
+    },
     "storage": {
       "patterns": [
         {
@@ -44,28 +51,12 @@
       ]
     },
     "strings_interpolation": {
-      "name": "string.interpolated",
-      "begin": "m(%+)\"$",
-      "end": "\"\\1$",
+      "name": "meta.embedded",
+      "begin": "%\\{",
+      "end": "\\}",
       "patterns": [
         {
-          "name": "constant.character.escape.interpolation.nickel",
-          "match": "\\\\1$\\{"
-        },
-        {
-          "name": "constant.other.placeholder.nickel",
-          "begin": "\\1\\{",
-          "beginCaptures": {
-            "0": {
-              "name": "punctuation.section.curly.begin.nickel"
-            }
-          },
-          "end": "\\}",
-          "endCaptures": {
-            "0": {
-              "name": "punctuation.section.curly.end.nickel"
-            }
-          }
+          "include": "#expression"
         }
       ]
     },
@@ -85,6 +76,20 @@
         }
       ]
     },
+    "multiline_strings": {
+      "name": "string.quoted.double.nickel",
+      "begin": "m(%+)\"",
+      "end": "\"\\1",
+      "patterns": [
+        {
+          "name": "constant.character.escape.nickel",
+          "match": "\\\\."
+        },
+        {
+          "include": "#strings_interpolation"
+        }
+      ]
+    },
     "strings": {
       "name": "string.quoted.double.nickel",
       "begin": "\"",
@@ -93,6 +98,9 @@
         {
           "name": "constant.character.escape.nickel",
           "match": "\\\\."
+        },
+        {
+          "include": "#strings_interpolation"
         }
       ]
     },
@@ -137,7 +145,7 @@
       "patterns": [
         {
           "name": "keyword.operator",
-          "match": "(=>|==|=|\\|\\||\\||->|<|>|>=|<=|\\+\\+|\\+|\\-\\$|\\-|\\*|\\/|&&|&|\\.|\\:|@)"
+          "match": "(=>|==|=|\\|\\||\\||->|<|>|>=|<=|\\+\\+|\\+|\\-\\$|\\-|\\*|\\/|&&|&|\\.|\\:|@|')"
         }
       ]
     }
```

File: lsp/vscode-extension/language-configuration.json (hunk 1 above)
Comment (yannham) [reply lsp/vscode-extension/language-configuration.json:21]: You're right, but it should rather be an separate rule, I guess. In the vim extension, if you type `'foo`, the whole word + starting quote is getting colored in a special way, which is probably the desired behavior. Let met try to add that.

File: lsp/vscode-extension/syntaxes/nickel.tmLanguage.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (yannham) [reply lsp/vscode-extension/syntaxes/nickel.tmLanguage.json:57]: Ah, I think what we are missing in order to reuse `\1` is `beginCaptures`, after a quick look at TextMate's documentation

[the remark this answers is not in the document]
Comment (yannham) [reply lsp/vscode-extension/syntaxes/nickel.tmLanguage.json:57]: Anyway, for a fist version, we can very much start with support just for a single delimiter.

**Comment by yannham** [comment]
@vkleen , If you're willing to review the code, I tested the extension with VSCode on latest commit ([771f38c](https://github.com/tweag/nickel/pull/1413/commits/771f38c90a9ab8eb99049288adf7a27f1afecb34)) and everything seems to work fine (up to a small fix of removing an `unwrap` in the LSP, but I'll do that as a separate PR to backport it onto `release-1.1.0`).

@szlend Thanks again for your help! It seems to work fine :+1:  this version doesn't correctly handle interpolation sequences with longer delimiters than `m%"`, but that would do for now.

**Review by vkleen — APPROVED** [review]
Looks reasonable to me, with the caveat that I haven't tested it myself.

**Pull request merged**